src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/MavenUtils.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            pomLocation = userDefinedLocalRepositoryPath;
+        }
+
+        var componentName = scannedComponent.Component.PackageUrl?.Name.ToLower();
+        var componentNamespace = scannedComponent.Component.PackageUrl?.Namespace?.ToLower();
+        var componentVersion = scannedComponent.Component.PackageUrl?.Version;
+
+        // Take the component namespace and split it by "." in order to get the correct path to the .pom
+        if (!string.IsNullOrEmpty(componentNamespace))
+        {
+            var componentNamespaceParts = componentNamespace.Split('.'); // Example: "org.apache.commons.commons-lang3"
+
+            for (var i = 0; i < componentNamespaceParts.Length; i++)
+            {
+                pomLocation += $"/{componentNamespaceParts[i]}";
+            }
+        }
+
+        pomLocation += $"/{componentName}/{componentVersion}/{componentName}-{componentVersion}.pom";
+
+        // Check for file permissions on the .m2 directory before attempting to check if the file exists
+        if (MavenPackagesPathHasReadPermissions)
+        {
+            if (fileSystemUtils.FileExists(pomLocation))
+            {
+                return pomLocation;
+            }
+            else
+            {
+                log.Debug($"Pom location could not be found at: {pomLocation}");
+            }
+        }
+
+        return null;
+    }
+
+    public (string Name, string Version, PackageDetails packageDetails) ParsePom(string pomLocation)
+    {
+        var supplierField = string.Empty;
+        var licenseField = string.Empty;
+
+        try
+        {
+            var pomBytes = fileSystemUtils.ReadAllBytes(pomLocation);
+            using var pomStream = new MemoryStream(pomBytes, false);
+
+            var doc = new XmlDocument();
+            doc.Load(pomStream);
+
+            XmlNode developersNode = doc["project"]?["developers"];
+            XmlNode licensesNode = doc["project"]?["licenses"];
+            XmlNode organizationNode = doc["project"]?["organization"];
+
+            var name = doc["project"]?["artifactId"]?.InnerText;
+            var version = doc["project"]?["version"]?.InnerText;
+
+            if (organizationNode != null)
+            {
+                var organizationName = organizationNode["name"]?.InnerText;
+                if (!string.IsNullOrEmpty(organizationName))
+                {
+                    supplierField = $"Organization: {organizationName}";
+                }
+            }
+            else if (developersNode != null)
+            {
+                // Take the first developer name and use it as the supplier when there is no organization listed.
+                var developerName = developersNode["developer"]?["name"]?.InnerText;
+                if (!string.IsNullOrEmpty(developerName))
+                {
+                    supplierField = $"Person: {developerName}";
+                }
+            }
```

Comment:
Should we have a final case here, perhaps at least a debug for "We don't know who produced this"?

Reply:
At the Generator.cs there is a check for when this is null it gets populated as "NOASSERTION" https://github.com/microsoft/sbom-tool/blob/main/src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Generator.cs#L164 This also applies to other fields such as LicenseDeclared and LicenseConcluded